docs/howtos/ext-filters.md

```diff
@@ -0,0 +1,170 @@
+    targetPort: http-api
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-auth
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: example-auth
+  template:
+    metadata:
+      labels:
+        app: example-auth
+    spec:
+      containers:
+      - name: example-auth
+        image: docker.io/datawire/ambassador-auth-service:2.0.0
+        imagePullPolicy: Always
+        ports:
+        - name: http-api
+          containerPort: 3000
+        resources:
+          limits:
+            cpu: "0.1"
+            memory: 100Mi
+```
+
+Note that the cluster does not yet contain any Ambassador AuthService definition. This is intentional: we want the service running before we tell Ambassador about it.
+
+The YAML above is published at getambassador.io, so if you like, you can just do
+
+```shell
+kubectl apply -f https://www.getambassador.io/yaml/demo/demo-auth.yaml
+```
+
+to spin everything up. (Of course, you can also use a local file, if you prefer.)
+
+Wait for the pod to be running before continuing. The output of `kubectl get pods` should look something like
+
+```console
+$ kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+example-auth-6c5855b98d-24clp   1/1       Running   0          4m
+```
+Note that the `READY` field says `1/1` which means the pod is up and running.
+
+## 2. Configure Ambassador Authentication
+
+Once the auth service is running, we need to tell Ambassador about it. The easiest way to do that is to map the `example-auth` service with the following:
+
+```yaml
+---
+apiVersion: getambassador.io/v2
+kind: Filter
+metadata:
+  name: authentication
+spec:
+  External:
+    auth_service: "example-auth:3000"
+    path_prefix: "/extauth"
+    allowed_request_headers:
+    - "x-qotm-session"
+    allowed_authorization_headers:
+    - "x-qotm-session"
+```
+
+This configuration tells Ambassador about the filter, notably that it needs the `/extauth` prefix, and that it's OK for it to pass back the `x-qotm-session` header. Note that `path_prefix` and `allowed_headers` are optional.
+
+If the auth service uses a framework like [Gorilla Toolkit](http://www.gorillatoolkit.org) which enforces strict slashes as HTTP path separators, it is possible to end up with an infinite redirect where the filter's framework redirects any request with non-conformant slashing. This would arise if the above example had ```path_prefix: "/extauth/"```, the filter would see a request for ```/extauth//backend/get-quote/``` which would then be redirected to ```/extauth/backend/get-quote/``` rather than actually be handled by the authentication handler. For this reason, remember that the full path of the incoming request including the leading slash, will be appended to ```path_prefix``` regardless of non-conformant slashing.
+
+## 3. Test Authentication
+
+If we `curl` to a protected URL:
+
+```shell
+$ curl -Lv $AMBASSADORURL/backend/get-quote/
```

Comment:
Re: My comment about `FilterPolicy` above.

Alternatively, you could have the docs show how cool™ it is that you can select which requests get authenticated.

You could have the `FilterPolicy` only run the `Filter` on requests to `path: /backend/get-quote/` to show how you can have authentication on different routes.

```yaml
---
apiVersion: getambassador.io/v2
kind: FilterPolicy
metadata:
  name: authentication
spec:
  rules:
  - host: "*"
    path: /backend/get-quote/
    filters:
    - authentication
```

Reply:
I added both `FilterPolicy` examples, first with all hosts/path, then with the how cool™ example, hows's that look?